programs/ssl/ssl_test_lib.c

```diff
@@ -0,0 +1,283 @@
+        min_version = ciphersuite_info->min_minor_ver;
+        /* DTLS starts with TLS 1.1 */
+        if( transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM &&
+            min_version < MBEDTLS_SSL_MINOR_VERSION_2 )
+            min_version = MBEDTLS_SSL_MINOR_VERSION_2;
+    }
+
+    /* Enable RC4 if needed and not explicitly disabled */
+    if( ciphersuite_info->cipher == MBEDTLS_CIPHER_ARC4_128 )
+    {
+        if( *arc4 == MBEDTLS_SSL_ARC4_DISABLED )
+        {
+            mbedtls_printf("forced RC4 ciphersuite with RC4 disabled\n");
+        return( 2 );
+        }
+
+        *arc4 = MBEDTLS_SSL_ARC4_ENABLED;
+    }
+    return( 0 );
+}
+
+#if defined(MBEDTLS_X509_CRT_PARSE_C)
+const int mbedtls_ssl_test_sig_hashes_with_sha1[] = {
+#if defined(MBEDTLS_SHA512_C)
+    MBEDTLS_MD_SHA512,
+    MBEDTLS_MD_SHA384,
+#endif
+#if defined(MBEDTLS_SHA256_C)
+    MBEDTLS_MD_SHA256,
+    MBEDTLS_MD_SHA224,
+#endif
+#if defined(MBEDTLS_SHA1_C)
+    /* Allow SHA-1 as we use it extensively in tests. */
+    MBEDTLS_MD_SHA1,
+#endif
+    MBEDTLS_MD_NONE
+};
+
+void mbedtls_ssl_test_conf_allow_sha1( mbedtls_ssl_config *conf,
+                                       mbedtls_x509_crt_profile *crt_profile )
+{
+    *crt_profile = mbedtls_x509_crt_profile_default;
+    crt_profile->allowed_mds |= MBEDTLS_X509_ID_FLAG( MBEDTLS_MD_SHA1 );
+    mbedtls_ssl_conf_cert_profile( conf, crt_profile );
+    mbedtls_ssl_conf_sig_hashes( conf, mbedtls_ssl_test_sig_hashes_with_sha1 );
+}
+#endif /* MBEDTLS_X509_CRT_PARSE_C */
+
+#if defined(MBEDTLS_ECP_C)
+int mbedtls_ssl_test_parse_curves( char *p,
+                                 mbedtls_ecp_group_id *curve_list )
+{
+    int i = 0;
+    const mbedtls_ecp_curve_info *curve_cur;
+    const char *q;
+
+    if( p == NULL )
+        return( 0 );
+
+    if( strcmp( p, "none" ) == 0 )
+    {
+        curve_list[0] = MBEDTLS_ECP_DP_NONE;
+    }
+    else if( strcmp( p, "default" ) != 0 )
+    {
+        /* Leave room for a final NULL in curve list */
+        while( i < CURVE_LIST_SIZE - 1 && *p != '\0' )
+        {
+            q = p;
+
+            /* Terminate the current string */
+            while( *p != ',' && *p != '\0' )
+                p++;
+            if( *p == ',' )
+                *p++ = '\0';
+
+            if( ( curve_cur = mbedtls_ecp_curve_info_from_name( q ) ) != NULL )
+            {
+                curve_list[i++] = curve_cur->grp_id;
+            }
+            else
+            {
+                mbedtls_printf( "unknown curve %s\n", q );
+                mbedtls_printf( "supported curves: " );
+                for( curve_cur = mbedtls_ecp_curve_list();
+                     curve_cur->grp_id != MBEDTLS_ECP_DP_NONE;
+                     curve_cur++ )
+                {
+                    mbedtls_printf( "%s ", curve_cur->name );
+                }
+                mbedtls_printf( "\n" );
+                return( 2 );
+            }
+        }
+
+        mbedtls_printf("Number of curves: %d\n", i );
```

Comment:
Probably not your fault, but our style says we need more spaces around parenthesis here.

```diff
- mbedtls_printf("Numb
+ mbedtls_printf( "Numb
```

Reply:
Out of scope, but I did a search-and-replace anyway.